src/graph.jl

```diff
@@ -72,22 +86,46 @@ function BaseGraph(filepath::AbstractString,
     end
 
     # get attributes
-    attributes = get_attributes(raw_graph, num_nodes, nodes_str)
+    attributes = get_attributes(nodes)
 
     return BaseGraph(num_nodes, num_edges, num_districts, total_pop,
                      populations, adj_matrix, edge_src, edge_dst, neighbors,
                      simple_graph, attributes)
 end
 
 
-function get_attributes(raw_graph::Dict{String, Any},
-                        num_nodes::Int,
-                        nodes_str::AbstractString = "nodes")
+function BaseGraph(filepath::AbstractString,
+                   pop_col::AbstractString,
+                   assignment_col::AbstractString)::BaseGraph
+    """ Builds the base Graph object. This is the underlying network of our
+        districts, and its properties are immutable i.e they will not change
+        from step to step in our Markov Chains.
+
+        Arguments:
+        filepath:       A path to a .json or .shp file which contains the
+                        information needed to construct the graph.
+        pop_col:        the node attribute key whose accompanying value is the
+                        population of that node
+        assignment_col: the node attribute key whose accompanying value is the
+                        assignment of that node (i.e., to a district)
+    """
+    extension = splitext(filepath)[2]
+    if extension == ".json"
+        return graph_from_json(filepath, pop_col, assignment_col)
+    elseif extension == ".shp"
```

Comment:
Strict extension filtering makes me a little queasy. For instance, IIRC a lot of old shapefiles have a `.SHP` extension (🎶 back in the '90s... 🎶), rather than a `.shp` extension. You could just make `extension` lowercase. However, I would consider using a try/catch block here: first try to read as JSON, then try to read as a shapefile, and then reject if both fail. This way, you don't have to think about all the weird things that can happen with filenames.

Reply:
An observation: GerryChain allows for the loading of a shapefile without an explicit `.shp` extension. For instance, `Graph.from_file('PA')` is equivalent`Graph.from_file('PA/PA.shp')`. Because it has to guess what the actual shapefile within the directory is, this can lead to some strange behavior, so I suggest we **don't** support this. May be worth documenting the difference, though.

Reply:
will document! and also will make sure to add the change with the `.json` and `.shp`, that's a great idea!